### PR TITLE
Make sure we can use PSR-18 clients in version 1.x

### PR DESCRIPTION
**Pull request by Nyholm**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| Documentation   | 
| License         | MIT

Our 2.x branch is compatible with PSR-18. The master branch should also be compatible. 

Ive ipdated all places where we are **consuming** a httplug client. They do now support both httplug 1.x and PSR-18.

FYI @gmponos

### CODE REVIEW

File: src/FlexibleHttpClient.php [hunk 1]
```diff
@@ -21,14 +22,14 @@
      */
```
Comment (GrahamCampbell) [inline src/FlexibleHttpClient.php]: param doc needs updating here i think?

**Comment by gmponos** [comment]
Just throwing a long shot idea here.. on v1 what if there was a PSR18 HTTPlug adapter?

I think someone else has also suggested this but I am lost with all these issues..

**Comment by Nyholm** [comment]
That would not help. Because PSR-18 has the exact same signature but with PHP7 return values. 

That is why all these updates to 2.x are super simple. They are just dropping PHP5 and adding return values all over the place.

File: src/FlexibleHttpClient.php (hunk 1 above)
Comment (Nyholm) [reply src/FlexibleHttpClient.php]: Thank you

**Comment by Nyholm** [comment]
Thank you for merging

**Pull request merged**